src/session/session_api.c

```diff
@@ -1475,6 +1476,24 @@ __wt_session_range_truncate(
               session, EINVAL, "the start cursor position is after the stop cursor position");
     }
 
+    /*
+     * Utilize the cursor lower bound as the temporary buffer to storing the original start and stop
+     * key. We use the original keys for the write-ahead log.
+     */
+    if (start != NULL) {
```

Comment:
I think there is one case that is interestingly different with this change. Previously it wasn't possible to have an empty start, and valid end by the time the log record was generated. Since the code further down in this function creates a start implicitly if there is a stop (line 1529).

Can you check to make sure this code does something sensible in that case?

i.e: a call like: `session->truncate(session, NULL, NULL, stop);`

I'm also a bit concerned that the code above creates and positions a start cursor implicitly if the URI is non-NULL.

Reply:
I have looked into the case for `session->truncate(session, NULL, NULL, stop);` and have found that we run into an assert that gets later hit because the start is NULL at this line of code. At line 1529, we implicitly create a start but we do not record the starting key. 

Similarly for `session->truncate(session, uri, NULL, NULL);` works properly but also has a problem. The start cursor gets implicitly and performs at next to position at the first record. Later on since we perform a get_key() it will return the first record's key instead of the original key. 

I will work for a solution that solves both these problems

Reply:
I have fixed this in [2d495e3](https://github.com/wiredtiger/wiredtiger/pull/8383/commits/2d495e34998743b1612a881b2df4fec9302dd313) let me know what you think!